### PR TITLE
Update RDKit versions used in tests

### PR DESCRIPTION
**Pull request by mattwthompson**
Work around

* https://github.com/rdkit/rdkit/issues/7890
* https://github.com/rdkit/rdkit/issues/7221 (and a few other stub-related issues)
* Some changes somewhere in 2023.3

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -76,7 +76,6 @@ jobs:
       - name: Install environment with ${{ env.JOBNAME }}
         uses: mamba-org/setup-micromamba@v1
         with:
-          micromamba-version: '1.5.10-0'
```
Comment (mattwthompson) [inline .github/workflows/CI.yml:79]: These are duplicate lines from a bad merge somewhere and were causing syntax errors

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -141,7 +140,10 @@ jobs:
 
       - name: Run mypy
         # subtle differences in Python/mypy versions, just keep it passing on one
-        if: ${{ matrix.rdkit == true && matrix.openeye && matrix.python-version == 3.10 }}
+        # prefer having htis back to rdkit == true, but builds often have bugs in stubs and
+        # there are other bugs/subtleties (in the source code and builds) that reduce the number
+        # of available builds
+        if: ${{ matrix.rdkit == false && matrix.openeye && matrix.python-version == 3.10 }}
```
Comment (mattwthompson) [inline .github/workflows/CI.yml:146]: Like it says - a lot of RDKit conda packages have bugs in their stubs that prevent type-checking, and type-checking only in the `rdkit == false` environments is better than nothing

File: openff/toolkit/_tests/test_toolkits.py [hunk 3]
```diff
@@ -193,8 +193,6 @@ def get_mini_drug_bank(toolkit_class, xfail_mols=None):
     "DrugBank_1962",
     "DrugBank_5043",
     "DrugBank_2519",
-    "DrugBank_7124",
-    "DrugBank_6865",
```
Comment (mattwthompson) [inline openff/toolkit/_tests/test_toolkits.py:197]: I can't find the relevant change in the RDKit changelog, but this basically reverts part of https://github.com/openforcefield/openff-toolkit/pull/895

**Comment by mattwthompson** [comment]
Sure - I think it might come down to why the behavior around https://github.com/openforcefield/openff-toolkit/pull/1942#discussion_r1786378162 changed in some of the matrix but not the rest

**Comment by j-wags** [comment]
Cool, I'll take this over then!

**Comment by j-wags** [comment]
The molecules that are newly failing (er, passing) stereo tests are somewhat exotic rings with nitrogens. Loading both with RDKit 2022 leads to undefined stereo errors on nitrogens[1], but not any more in RDKit 2024. Without fully delving into the context of all our past decisions about interop, raising fewer stereo errors here is a GOOD thing, since _the molecules are 3D to begin with_, so the idea that they're missing stereo is nonsensical. So I have no hesitation removing molecules from this xfail list. 
 
Screenshot 2024-10-04 at 2 07 39 PM

Screenshot 2024-10-04 at 2 07 07 PM

[1]  
```
openff.toolkit.utils.exceptions.UndefinedStereochemistryError: Unable to make OFFMol from RDMol: RDMol has unspecified stereochemistry. RDMol name: DrugBank_7124Undefined chiral centers are:
 - Atom N (index 43)
```

**Comment by j-wags** [comment]
I'll consider my final round of cleanups to be equivalent to my review and will merge this PR once I've made them.

**Review by j-wags — APPROVED** [review]

**Comment by mattwthompson** [comment]
Great - removing skipped molecules meshes with my surface-level understanding of those "failures" being improvements

**Pull request merged**